airbyte-cdk/python/airbyte_cdk/sources/streams/http/exceptions.py

```diff
@@ -9,7 +9,11 @@
 
 
 class BaseBackoffException(requests.exceptions.HTTPError):
-    pass
+    def __init__(self, request: requests.PreparedRequest, response: requests.Response):
+        self.request = request
+        self.response = response
+        message = f"\nRequest URL: {self.request.url}\nResponse Code: {self.response.status_code}\nResponse Text: {self.response.text}"
+        super().__init__(message)
```

Comment:
```suggestion
    def __init__(self, request: requests.PreparedRequest, response: requests.Response):
        error_message = f"\nRequest URL: {self.request.url}\nResponse Code: {self.response.status_code}\nResponse Text: {self.response.text}"
        super().__init__(error_message, request=request, response=response)
```
According to the [parent class implementation](https://github.com/psf/requests/blob/fa1b0a367abc8488542f7ce7c02a3614ad8aa09d/requests/exceptions.py#L14) the request/response should be passed as kwargs. This means you don't need to set these instance attributes, the parent class will take care of this if you pass these in the `super().__init__`.